diff --git a/src/offline_viewer.cpp b/src/offline_viewer.cpp
index 4099acb..c27ff37 100644
--- a/src/offline_viewer.cpp
+++ b/src/offline_viewer.cpp
@@ -57,6 +57,6 @@ int main(int argc, char** argv) {
     spdlog::info("map_path={}", init_map_path);
   }
 
-  glim::OfflineViewer viewer(init_map_path, vm["export_path"].as<std::string>());
+  glim::OfflineViewer viewer(init_map_path);
   viewer.wait();
 }
\ No newline at end of file
